### PR TITLE
- Fixes zonemaster/zonemaster-engine#543

### PR DESCRIPTION
**Pull request by vlevigneron**
- Fixes zonemaster/zonemaster-engine#543
- CLI "no-ipv4" and "no-ipv6" options are now taken into account
- Should be merged before PR zonemaster/zonemaster-engine#586

### CODE REVIEW

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 1]
```diff
@@ -1460,6 +1462,22 @@ sub dnssec11 {
     return @results;
 } ## end sub dnssec11
 
+sub _is_ip_version_disabled {
+    my $ns = shift;
+
+    if ( not Zonemaster::Engine::Profile->effective->get(q{net.ipv4}) and $ns->address->version == $IP_VERSION_4 ) {
+        Zonemaster::Engine->logger->add( SKIP_IPV4_DISABLED => { ns => $ns->name->string.q{/}.$ns->address->short } );
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: To be consistent with the other modules the message should be IPV4_DISABLED. Only System uses SKIP_IPV4_DISABLED. The message must also be added to profile.json with level DEBUG (only Basic has INFO).

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 2]
```diff
@@ -1460,6 +1462,22 @@ sub dnssec11 {
...
+        return 1;
+    }
+
+    if ( not Zonemaster::Engine::Profile->effective->get(q{net.ipv6}) and $ns->address->version == $IP_VERSION_6 ) {
+        Zonemaster::Engine->logger->add( SKIP_IPV6_DISABLED => { ns => $ns->name->string.q{/}.$ns->address->short } );
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: IPV6_DISABLED and see IPV4 above.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 1 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Well, SKIP_IPV4_DISABLED/IPV4_DISABLED needs some re-factoring. For instance SKIP_IPV4_DISABLED is already used in Zone.pm module. It is not the purpose of this issue/PR to fix this re-factoring. In the majority of cases IPV4_DISABLED is used so I'll modify DNSSEC.pm accordingly.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 2 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Same remark.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 3]
```diff
@@ -453,6 +455,12 @@ Readonly my %TAG_DESCRIPTIONS => (
     INVALID_NAME_RCODE => sub {           # INVALID_NAME_RCODE
         __x "When asked for the name {name}, which must not exist, the response had RCODE {rcode}.", @_;
     },
+    IPV4_DISABLED => sub {                  # IPV4_DISABLED
+        __x 'IPv4 is disabled, not sending "{rrtype}" query to {ns}/{address}.', @_;
+    },
+    IPV6_DISABLED => sub {                  # IPV6_DISABLED
+        __x 'IPv6 is disabled, not sending "{rrtype}" query to {ns}/{address}.', @_;
+    },
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: Following the new instructions by @mattias-p in PR #588 it should rather be

```
    IPV4_DISABLED => sub { 
        __x # IPV4_DISABLED
       'IPv4 is disabled, not sending "{rrtype}" query to {ns}/{address}.', @_;
    },
    IPV6_DISABLED => sub {
        __x # IPV6_DISABLED
        'IPv6 is disabled, not sending "{rrtype}" query to {ns}/{address}.', @_;
    },
```

Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: @matsduf  Just did the change. Please approve again.

**Pull request merged**